plugins/notifications-backend/src/service/types.ts

```diff
@@ -54,7 +48,7 @@ export const NotificationsOrderByDirections: string[] = ['asc', 'desc'];
  */
 export const MessageScopes = ['all', 'user', 'system'];
 
-export const DefaultUser = 'default/guest';
+export const DefaultServiceUser = 'default/externalServiceNotificationsUser';
```

Comment:
we create such a user? if yes then why? we can stay with the guest user.

Reply:
The original use-case for DefaultUser is gone, we do not support it. The identity of the user must be provided from the Backstage API, no matter it internally falls-back to the `default/user`. We newly throw exception if the user identity is missing.

The idea behind `DefaultServiceUser` is to differentiate an external caller from a regular user.
We are not there yet, but I am considering cases where 
- the external caller can retrieve messages of any user (if authorized)
- we store info about creator of the particular message

Reply:
1. the cases you describe are relevant for internal users as well. E.g. admin
2. let's not prepare too much for the future. if we'll need to differentiate something then we'll do the coding. so far we have just 'user' and 'default'. service is very vague.

Reply:
I still do not have a solution for the external caller flow.
I see a lot of people were struggling with this task during the past half a year, a lot of questions/issues raised, no relevant response provided.

Let's wait with this topic till we have complete vision how to identify/authorize the external caller.
Anyway, it would be wrong to reuse the default user for any external caller.

Reply:
if you can't support external users then remove it.
regarding 'Anyway, it would be wrong to reuse the default user for any external caller.'. it is not reusing. you have just one default

Reply:
We must support them. IIUC, the Orchestrator will be such an example of external services sending messages to Backstage. A workflow reaches certain step and they will send a notification. This is what I understood.

I have changed the workaround to a simple shared secret. I think it's good enough for now. The JWT token would be clean but I can not make it working. Will keep trying but I want to move forward as well.

Reply:
I think that it will be ok for now to work without authentication. For Orchestrator and any other client that wants to post notifications.
But whether we support it or not is not really our problem. it is part of the backstage installation.
and we still have just one default user for any source , external or internal or whatever.

Reply:
I disagree, we must support authentication. The feature recently allows sending messages to any user or their groups by simply reaching a REST endpoint.

Even the Orchestrator has raised this issue before. I do not remember them accepting the missing authentication, they just got muted. And it is a must-have from the upstream perspective.

Can you describe how the "backstage installation" would implement AA without collaboration with the plugin?
IIRC, we have been told to skip it for the reason you have mentioned. But with the experience we gained since those days, I do not see it as a correct solution.